spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -253,5 +295,77 @@
         expect(pdf_fields["ZIP Code"]).to eq "08037"
       end
     end
+
+    describe "filing status" do
+      context "single" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 1
+        end
+        it "checks the Choice1 box" do
+          expect(pdf_fields["Group1"]).to eq "Choice1"
+        end
+      end
+
+      context "married filing jointly" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 2
+        end
+        it "checks the Choice2 box" do
+          expect(pdf_fields["Group1"]).to eq "Choice2"
+        end
+      end
+
+      context "married filing separately" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 3
+        end
+        it "checks the Choice3 box" do
+          expect(pdf_fields["Group1"]).to eq "Choice3"
+        end
+      end
+
+      context "head of household" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 4
+        end
+        it "checks the Choice4 box" do
+          expect(pdf_fields["Group1"]).to eq "Choice4"
+        end
+      end
+
+      context "qualifying widow" do
+        context "spouse passed in the last year" do          
+          before do
+            submission.data_source.direct_file_data.filing_status = 5
+            date_within_prior_year = "#{MultiTenantService.new(:statefile).current_tax_year}-09-30"
+            submission.data_source.direct_file_data.spouse_date_of_death = date_within_prior_year
+          end
+  
+          it "checks the Choice5 box" do
+            expect(pdf_fields["Group1"]).to eq "Choice5"
+          end
+  
+          it "checks the one year prior spouse date of death" do
+            expect(pdf_fields["Group1qualwi5ab"]).to eq "1"
+          end
+        end
+
+        context "spouse passed two years prior" do          
+          before do
+            submission.data_source.direct_file_data.filing_status = 5
+            date_two_years_prior = "#{MultiTenantService.new(:statefile).current_tax_year - 1}-09-30"
+            submission.data_source.direct_file_data.spouse_date_of_death = date_two_years_prior
+          end
+  
+          it "checks the Choice5 box" do
+            expect(pdf_fields["Group1"]).to eq "Choice5"
+          end
+  
+          it "checks the two years prior spouse date of death" do
+            expect(pdf_fields["Group1qualwi5ab"]).to eq "0"
```

Comment:
[DUST] this says "checks the 2 years prior" box but is really just checking that the one year box is _not_ checked, so i'd update the description accordingly

Reply:
ohhh i've just read `nj1040_pdf.rb#spouse_death_year` and i understand now. nvm!